python/tvm/relax/block_builder.py

```diff
@@ -629,9 +624,7 @@ def emit_func_output(
         seqe = self.normalize(rx.SeqExpr(self._blocks, output))
 
         # The function's checked_type_ relies on the function body(seqe) to have deduced type
-        # TODO(@yuchen): handle the case where the body's checked_type_ is null
-        # TODO: Deduce the ret shape too
-        func = rx.Function(self._func_params, seqe, None, rx.RuntimeDepShape())
+        func = rx.Function(self._func_params, seqe, None)
```

Comment:
Are we able to set the struct info of this function now?
```suggestion
        func = rx.Function(self._func_params, seqe, seqe.struct_info)
```

Reply:
Not setting is probably fine as it will auto-deduce. Added a comment

Reply:
Okay I see the constructor takes an `Optional<StructInfo> ret_struct_info` as argument, so it’s fine. But for another thing, do you think it makes sense having a default value `None` or `NullOpt` for this argument in the constructor on both Python side and C++ side?
https://github.com/tlc-pack/relax/blob/82e54019d8af8c2076aa564b5a740ef7418a46f8/python/tvm/relax/expr.py#L342
https://github.com/tlc-pack/relax/blob/82e54019d8af8c2076aa564b5a740ef7418a46f8/include/tvm/relax/expr.h#L832

Given that the constructor will always deduce when no return structure info is given, I think we can probably have the default value.

Reply:
With the default value, we won’t need to explicitly pass `None` to the constructor here then.

Reply:
That makes sense, will update now